system/Boot.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter;
+
+use CodeIgniter\Cache\FactoriesCache;
+use CodeIgniter\CLI\Console;
+use CodeIgniter\Config\DotEnv;
+use CodeIgniter\Exceptions\FrameworkException;
+use Config\Autoload;
+use Config\Cache;
+use Config\Modules;
+use Config\Paths;
+use Config\Services;
+
+/**
+ * Bootstrap for the application
+ *
+ * @codeCoverageIgnore
+ */
+class Boot
+{
+    /**
+     * @used-by public/index.php
+     *
+     * Context
+     *   web:     Invoked by HTTP request
+     *   php-cli: Invoked by CLI via `php public/index.php`
+     *
+     * @return int Exit code.
+     */
+    public static function bootWeb(Paths $paths): int
+    {
+        static::loadDotEnv($paths);
+        static::defineEnvironment();
+        static::loadEnvironmentBootstrap($paths);
+        static::definePathConstants($paths);
+        if (! defined('APP_NAMESPACE')) {
+            static::loadConstants();
+        }
+        static::loadCommonFunctions();
+        static::loadAutoloader();
+        static::setExceptionHandler();
+        static::checkMissingExtensions();
+        static::initializeKint();
+
+        $configCacheEnabled = (new Cache())->configCacheEnabled ?? false;
```

Comment:
The `Cache` Config class that extends `BaseConfig` is very slow.

Add a new plain PHP class.
```php
<?php

namespace Config;

class Optimize
{
    /**
     * --------------------------------------------------------------------------
     * Config Caching
     * --------------------------------------------------------------------------
     *
     * @see https://codeigniter.com/user_guide/concepts/factories.html#config-caching
     */
    public bool $configCacheEnabled = false;
}
```

```php
        $iterator = new \CodeIgniter\Debug\Iterator();

        $iterator->add('Config\Cache', static function ($word = 'little') {
            $configCacheEnabled = (new Cache())->configCacheEnabled ?? false;
        });

        $iterator->add('Config\Optimize', static function ($word = 'little') {
            $configCacheEnabled = class_exists(Optimize::class) ? (new Optimize())->configCacheEnabled : false;
        });

        return $iterator->run(3000);
```

Results:
```
	Test 	Time 	Memory
config\cache 	0.1990 	0 Bytes
config\optimize	0.0006 	0 Bytes
```

```
$ php -v
PHP 8.2.16 (cli) (built: Feb 16 2024 05:30:16) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.16, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.16, Copyright (c), by Zend Technologies
$ sw_vers
ProductName:	macOS
ProductVersion:	12.7.3
BuildVersion:	21H1015
```